nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py

```diff
@@ -341,6 +341,33 @@ def run(config):
     address = os.environ.get("REDIS_ADDRESS", config.get("redis_address"))
     ray.init(address=address, local_mode=config.get("local_mode", False))
 
+    # Register serializer and deserializer - needed when logging arrays and tensors.
+    def serializer(obj):
+        if obj.requires_grad:
+            obj = obj.detach()
+        if obj.is_cuda:
+            return obj.cpu().numpy()
+        else:
+            return obj.numpy()
+
+    for tensor_type in [
+        torch.FloatTensor,
+        torch.DoubleTensor,
+        torch.HalfTensor,
+        torch.ByteTensor,
+        torch.CharTensor,
+        torch.ShortTensor,
+        torch.IntTensor,
+        torch.LongTensor,
+        torch.Tensor,
+    ]:
+        def deserializer(serialized_obj):
```

Comment:
Dont you need to know the destination device (cpu or gpu) when you deserialize?

Reply:
That's a good point. So far, I haven't seen this cause any issues, but that may be due to the limited application I've testing (with the logs).